src/embed.ts

```diff
@@ -366,6 +357,17 @@ async function _embed(
     ...(ast ? {expr: (vega as any).expressionInterpreter} : {}),
   });
 
+  view.addSignalListener('autosize', (_, autosize: Exclude<AutoSize, string>) => {
+    const {type} = autosize;
+    container.classList.remove('fit-x', 'fit-y');
+    if (type == 'fit' || type == 'fit-x') {
+      container.classList.add('fit-x');
+    }
+    if (type == 'fit' || type == 'fit-y') {
+      container.classList.add('fit-y');
```

Comment:
Looks good. Just curious, do you know how the browser prevents flickering from the rapid remove / add of classes? Assuming there is some mechanism that executes blocks of CSS updates?

Reply:
I don't think there are rapid updates, are there? I would expect a few updates in a while.